pkg/instances/instances.go

```diff
@@ -181,11 +198,145 @@ func (i *instances) Configuration(id string) string {
+					"app": label,
+				}
+				options.LabelSelector = labels.FormatLabels(labelSelector)
+
+				p, err := i.kubeClient.CoreV1().Pods(v1.NamespaceAll).List(options)
+				if err == nil && len(p.Items) == 1 {
+					pod := p.Items[0]
+
+					image := pod.Spec.Containers[0].Image
+					namespace := pod.GetNamespace()
+					age := age.GetAge(pod.CreationTimestamp.Time)
+					created := pod.CreationTimestamp.Format("2006-01-02 15:04.05")
+					version := image[strings.IndexAny(image, ":")+1:]
+					status := ""
+
+					if pod.Status.ContainerStatuses[0].State.Waiting != nil {
+						status = fmt.Sprintf("Waiting (%s)", pod.Status.ContainerStatuses[0].State.Waiting.Reason)
+					} else if pod.Status.ContainerStatuses[0].State.Running != nil {
+						status = "Running"
+					} else if pod.Status.ContainerStatuses[0].State.Terminated != nil {
+						status = "Terminated"
+					}
+
+					healthy := "False"
+					if pod.Status.ContainerStatuses[0].Ready {
+						healthy = "True"
+					}
+
+					s := StatusOutput{
+						Name:      label,
+						Namespace: namespace,
+						Created:   created,
+						Age:       age,
+						Status:    status,
+						Version:   version,
+						Healthy:   healthy,
+					}
+
+					m.Lock()
+					statuses = append(statuses, s)
+					m.Unlock()
+				}
+				wg.Done()
+			}(lbl)
+		}
+
+		wg.Wait()
+		return statuses
+	}
+	return []StatusOutput{}
+}
+
+// GetMetadata returns the result from the /v1.0/metadata endpoint
+func (i *instances) GetMetadata(id string) []MetadataActiveActorsCount {
+	url := ""
+	if i.kubeClient != nil {
+		resp, err := i.kubeClient.AppsV1().Deployments(meta_v1.NamespaceAll).List((meta_v1.ListOptions{}))
+		if err != nil || len(resp.Items) == 0 {
+			return []MetadataActiveActorsCount{}
+		}
+
+		for _, d := range resp.Items {
+			if d.Spec.Template.Annotations[daprEnabledAnnotation] != "" {
+				daprID := d.Spec.Template.Annotations[daprIDAnnotation]
+				if daprID == id {
+					pods, err := i.kubeClient.CoreV1().Pods(d.GetNamespace()).List(meta_v1.ListOptions{
+						LabelSelector: labels.SelectorFromSet(d.Spec.Selector.MatchLabels).String(),
+					})
+					if err != nil {
+						log.Println(err)
+						return []MetadataActiveActorsCount{}
+					}
+
+					if len(pods.Items) > 0 {
+						p := pods.Items[0]
+						url = fmt.Sprintf("http://%v:%v/v1.0/metadata", p.Status.PodIP, 3500)
+					}
+				}
+			}
+		}
+
+	} else {
+		port := i.GetInstance(id).HTTPPort
+		url = fmt.Sprintf("http://localhost:%v/v1.0/metadata", port)
+	}
+	if url != "" {
+
+		resp, err := http.Get(url)
+		if err != nil {
+			log.Println(err)
+			return []MetadataActiveActorsCount{}
+		}
+
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			log.Println(err)
+			return []MetadataActiveActorsCount{}
+		}
+
+		var data MetadataOutput
+		if err := json.Unmarshal(body, &data); err != nil {
+			log.Println(err)
+			return []MetadataActiveActorsCount{}
+		}
+		return data.Actors
```

Comment:
This GetMetadata method should return MetadataOutput type. Create another method GetActiveActorsCount( data MetadataOutput) that returns data.Actors slice of MetadataActiveActorsCount.